dattri/algorithm/base.py

```diff
@@ -117,104 +109,135 @@ def _set_full_train_data(self, dataloader: torch.utils.data.DataLoader) -> None:
         Args:
             dataloader (DataLoader): The dataloader for full training samples.
         """
-        # This function may be overrided by the subclass
+        # This function may be overridden by the subclass
         self.full_train_dataloader = dataloader
 
-    def generate_test_query(
+    def generate_test_rep(
         self,
-        index: int,
+        ckpt_idx: int,
         data: Tuple[torch.Tensor, ...],
     ) -> torch.Tensor:
-        """Calculating the query based on the test data.
+        """Getting the initial representations of the test data.
+
+        Inner product attributor calculates the inner product between the (transformed)
+        train representations and test representations. This function generates the
+        initial test representations.
 
-        Inner product attributor calculates the inner product between the
-        train query and the transformation of the test query. This function
-        calculates the test query based on the test data.
+        The default implementation calculates the gradient of the test loss with respect
+        to the parameter. Subclasses may override this function to calculate something
+        else.
 
         Args:
-            index (int): The index of the model parameters. This index
-                is used for ensembling of different trained model
-                parameters.
-            data (Tuple[Tensor]): The test data. Normally this is a tuple
-                of input data and target data, the number of items in the
-                tuple should be aligned in the target function. The tensors'
-                shape follows (1, batchsize, ...).
+            ckpt_idx (int): The index of the model checkpoints. This index
+                is used for ensembling different trained model checkpoints.
+            data (Tuple[Tensor]): The test data. Typically, this is a tuple
+                of input data and target data but the number of items in this
+                tuple should align with the corresponding argument in the
+                target function. The tensors' shape follows (1, batch_size, ...).
```

Comment:
I guess I can open another PR to fix this (at least try to make them aligned as much as possible)

Reply:
probably not a really big deal for now since users can follow our examples, but it might cause some problems since those unsqueeze stuff might not be trivial or default for users.